### PR TITLE
Support for dynamically added jump_to and jump_to_id links

### PR DESCRIPTION
**Pull request by e-kolpakov**

### CODE REVIEW

File: test/test.jquery-xblock.js [hunk 1]
```diff
@@ -182,17 +187,43 @@ describe('jquery-xblock', function() {
...
                 'edx_demo_course',
                 '38751697369040e39ec1d0403efbac96'
             ]);
         });
 
+        it('should process dynamically added jump_to links', function(done){
+            var href = "/courses/edX/Other_CourseX/edx_other_course/jump_to/location://edX/Other_CourseX/edx_other_course/vertical/1234567890abcdef1234567890abcdef";
+            var linkDom = $("<a id='dynamic_jump_to_link'>Another Link to unit 2</a>")
+            linkDom.attr('href', href);
+            $('.xblock').append(linkDom);  // mutation observer handlers should fire heres
```
Comment (bradenmacdonald) [inline test/test.jquery-xblock.js]: Very minor typo here and below: should say "should fire here" not "heres"

**Comment by mtyaka** [comment]
Are the mutation observers really needed? I may be missing something, but it looks like jQuery event delegation should work well enough to catch clicks on dynamically inserted links.

Instead of
```javascript
this.processJumpToLinks(root);
this.attachDOMMutationObservers(root);
```

you would do something like this on jquery-xblock initialization:
```javascript
root.on('click', 'a', function(evt) {
  if ($(this).attr('href').match(/a-long-regex-or-two/)) {
    evt.preventDefault();
    // parse course_id, etc. & emit xblock_jump event.
  }
});
```

**Comment by e-kolpakov** [comment]
@mtyaka thanks for the hint - they aren't. Actually it's much simpler to achieve this using event propagation. Updated Pr, please review

File: test/test.jquery-xblock.js [hunk 2]
```diff
@@ -182,17 +186,43 @@ describe('jquery-xblock', function() {
...
                 'edx_demo_course',
                 '38751697369040e39ec1d0403efbac96'
             ]);
         });
 
+        it('should process dynamically added jump_to links', function(done){
+            var href = "/courses/edX/Other_CourseX/edx_other_course/jump_to/location://edX/Other_CourseX/edx_other_course/vertical/1234567890abcdef1234567890abcdef";
+            var linkDom = $("<a id='dynamic_jump_to_link'>Another Link to unit 2</a>")
+            linkDom.attr('href', href);
+            $('.xblock').append(linkDom);  // mutation observer handlers should fire here
```
Comment (mtyaka) [inline test/test.jquery-xblock.js]: Outdated comment - no longer using mutation observers.

File: test/test.jquery-xblock.js [hunk 3]
```diff
@@ -182,17 +186,43 @@ describe('jquery-xblock', function() {
...
+            setTimeout(function(){
```
Comment (mtyaka) [inline test/test.jquery-xblock.js]: The `setTimeout` probably isn't needed anymore.

**Comment by mtyaka** [comment]
Two minor comments in the test file, :+1:  when resolved.

File: jquery.xblock.js [hunk 4]
```diff
@@ -184,17 +183,13 @@
...
-                    $(linkDOM).on('click', function(e) {
-                        e.preventDefault();
-                        console.log(link_found.course_id, link_found.block_type, link_found.block_id);
-                        $(linkDOM).trigger('xblock_jump', [link_found.course_id, link_found.block_type, link_found.block_id]);
-                    });
-                    $(linkDOM).addClass('xblock-jump');
+                    evt.preventDefault();
+                    console.log(link_found.course_id, link_found.block_type, link_found.block_id);
+                    $(this).trigger('xblock_jump' +
+                    '', [link_found.course_id, link_found.block_type, link_found.block_id]);
```
Comment (mtyaka) [inline jquery.xblock.js]: The `+ ''` was probably left here by accident :)

**Comment by e-kolpakov** [comment]
@mtyaka thanks, good catch ;)

**Pull request merged**